deepin desktop environment: v20 -> v23Beta

wineee opened the pull request:
## Description of changes

deepin V23 Beta2 is officially [released](https://www.deepin.org/en/deepin-v23-beta2-released/) , follow to upgrade DDE

- dtk 5.6.10 -> 5.6.17,   new library [dtkdeclarative](https://github.com/linuxdeepin/dtkdeclarative/#dtkdeclarative), a toolkit based on QtQuick/QtQml
- All Go packages support go mod (buildGoModule) now, buildGoPackage is no longer used
- deepin-music/deepin-image-viewer/deepin-album/dde-launch(pad) refactor using qml (originally qtwidget)








<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->




![image](https://github.com/NixOS/nixpkgs/assets/22803888/39accb18-0440-4a50-964f-57c301e0a706)

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

wineee committed:
dtkcommon: 5.6.9 -> 5.6.17

wineee committed:
dtkcore: 5.6.10 -> 5.6.17

wineee committed:
dtkgui: 5.6.10 -> 5.6.17

wineee committed:
dtkwidget: 5.6.10 -> 5.6.17

wineee committed:
dtkdeclarative: init at 5.6.17

wineee committed:
qt5integration: 5.6.6 -> 5.6.17

wineee committed:
dwayland: 5.24.3-deepin.1.4 -> 5.25.0

wineee committed:
qt5platform-plugins: 5.6.9 -> 5.6.16

wineee committed:
dde-qt-dbus-factory: 5.5.22 -> 6.0.0

wineee committed:
deepin-pdfium: 1.0.1 -> 1.0.2

wineee committed:
util-dfm: 1.2.12 -> 1.2.16

wineee committed:
image-editor: 1.0.32 -> 1.0.35

wineee committed:
deepin-kwin: 5.24.3-deepin.1.9 -> 5.25.11

wineee committed:
dde-kwin: remove

wineee committed:
deepin-pw-check: 5.1.18 -> 6.0.2

wineee committed:
dde-api: 5.5.32 -> 6.0.7

wineee committed:
deepin-desktop-schemas: 5.10.11 -> 6.0.5

wineee committed:
startdde: 5.10.1 -> 6.0.10

wineee committed:
dde-daemon: 5.14.122 -> 6.0.22

wineee committed:
dde-account-faces: 1.0.12.1 -> 1.0.15

wineee committed:
deepin-wallpapers: 1.7.10 -> 1.7.16

wineee committed:
deepin-desktop-theme: init at 1.0.8

wineee committed:
dde-file-manager: 6.0.23 -> 6.0.31

wineee committed:
dde-dock: 5.5.81 -> 6.0.22

wineee committed:
dde-clipboard: 5.4.25 -> 6.0.7

wineee committed:
dde-control-center: 5.6.3 -> 6.0.28

wineee committed:
dde-app-services: 0.0.20 -> 1.0.23

wineee committed:
dde-polkit-agent: 5.5.22 -> 6.0.5

wineee committed:
dde-session-shell: 5.6.4 -> 6.0.10

wineee committed:
dde-session-ui: 5.6.2 -> 6.0.10

wineee committed:
dde-network-core: 1.1.8 -> 2.0.15

wineee committed:
dde-widgets: init at 6.0.14

wineee committed:
dde-calendar: 5.10.1 -> 5.11.0

wineee committed:
dpa-ext-gnomekeyring: 5.0.11 -> 6.0.1

wineee committed:
dde-launchpad: init at 0.0.2

wineee committed:
deepin-service-manager: init at 1.0.3

wineee committed:
dde-session: init at 1.1.9

wineee committed:
dde-appearance: init at 1.1.6

wineee committed:
deepin-picker: 5.0.28 -> 6.0.0

wineee committed:
dde-application-manager: init at 1.0.19

wineee commented:
@ofborg build deepin

wineee committed:
deepin-voice-note: 5.11.1 -> 6.0.13

wineee committed:
deepin-music: 6.2.31 -> 7.0.3

wineee committed:
deepin-album: 5.10.9 -> 6.0.2

wineee committed:
deepin-movie-reborn: 5.10.28 -> 6.0.5

wineee committed:
deepin-ocr-plugin-manager: init at unstable-2023-07-10

wineee committed:
deepin-image-viewer: 5.9.13 -> 6.0.2

wineee committed:
deepin-screensaver: 5.0.16 -> 5.0.18

wineee committed:
aliases: add dde-kwin

wineee committed:
services.deepin.app-services: add systemd service

https://github.com/linuxdeepin/dde-app-services/pull/45

wineee committed:
deepin: add v23 packages

wineee commented:
@ofborg build deepin

wineee commented:
@ofborg test deepin

wineee committed:
deepin-system-monitor: 5.9.33 -> 6.0.8

wineee committed:
dde-gsettings-schemas: remove outdata packages

wineee committed:
deepin-screen-recorder: 5.12.1 -> unstable-2023-07-10

wineee committed:
deepin-camera: 6.0.2 -> unstable-2023-09-26

wineee committed:
deepin: don't wait dde-wm-chooser in v23

wineee committed:
dde-gsettings-schemas: add gsettings-desktop-schemas

wineee commented:
@ofborg test deepin

wineee commented:
@ofborg test deepin

wegank commented on pkgs/desktops/deepin/go-package/deepin-desktop-schemas/default.nix:
```diff
@@ -1,25 +1,22 @@
 
-  nativeBuildInputs = [ glib ];
-  buildInputs = [ go-lib ];
+  vendorSha256 = "sha256-q6ugetchJLv2JjZ9+nevUI0ptizh2V+6SByoY/eFJJQ=";
```
```suggestion
  vendorHash = "sha256-q6ugetchJLv2JjZ9+nevUI0ptizh2V+6SByoY/eFJJQ=";
```

wegank commented on pkgs/desktops/deepin/go-package/deepin-pw-check/default.nix:
```diff
@@ -16,20 +12,18 @@
   };
 
-  goDeps = ./deps.nix;
+  vendorSha256 = "sha256-L0vUEkUN70Hrx5roIvTfaZBHbbq7mf3WpQJeFAMU5HY=";
```
```suggestion
  vendorHash = "sha256-L0vUEkUN70Hrx5roIvTfaZBHbbq7mf3WpQJeFAMU5HY=";
```

wegank commented on pkgs/desktops/deepin/go-package/dde-daemon/default.nix:
```diff
@@ -32,61 +28,56 @@
     (substituteAll {
-      src = ./0005-fix-custom-wallpapers-path.diff;
-      inherit coreutils;
+      src = ./0003-aviod-use-hardcode-path.diff;
```
```suggestion
      src = ./0003-avoid-use-hardcode-path.diff;
```

Typo.

NickCao commented:
@ofborg test deepin

wineee commented:
@ofborg test deepin

wineee commented:
@ofborg test deepin

wineee committed:
deepin-compressor: fix build for new dtk

wineee commented:
@ofborg test deepin

wineee commented:
@ofborg test deepin

pbsds commented:
Result of `nixpkgs-review pr 257400` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>deepin.deepin-clone</li>
    <li>deepin.deepin-reader</li>
  </ul>
</details>
<details>
  <summary>72 packages built:</summary>
  <ul>
    <li>deepin.dde-account-faces</li>
    <li>deepin.dde-api</li>
    <li>deepin.dde-app-services</li>
    <li>deepin.dde-appearance</li>
    <li>deepin.dde-application-manager</li>
    <li>deepin.dde-calendar</li>
    <li>deepin.dde-clipboard</li>
    <li>deepin.dde-control-center</li>
    <li>deepin.dde-control-center.dev</li>
    <li>deepin.dde-daemon</li>
    <li>deepin.dde-device-formatter</li>
    <li>deepin.dde-dock</li>
    <li>deepin.dde-dock.dev</li>
    <li>deepin.dde-file-manager</li>
    <li>deepin.dde-gsettings-schemas</li>
    <li>deepin.dde-launcher</li>
    <li>deepin.dde-launchpad</li>
    <li>deepin.dde-network-core</li>
    <li>deepin.dde-polkit-agent</li>
    <li>deepin.dde-qt-dbus-factory</li>
    <li>deepin.dde-session</li>
    <li>deepin.dde-session-shell</li>
    <li>deepin.dde-session-shell.dev</li>
    <li>deepin.dde-session-ui</li>
    <li>deepin.dde-widgets</li>
    <li>deepin.deepin-album</li>
    <li>deepin.deepin-calculator</li>
    <li>deepin.deepin-camera</li>
    <li>deepin.deepin-compressor</li>
    <li>deepin.deepin-desktop-schemas</li>
    <li>deepin.deepin-desktop-theme</li>
    <li>deepin.deepin-draw</li>
    <li>deepin.deepin-editor</li>
    <li>deepin.deepin-image-viewer</li>
    <li>deepin.deepin-kwin</li>
    <li>deepin.deepin-kwin.dev</li>
    <li>deepin.deepin-movie-reborn</li>
    <li>deepin.deepin-movie-reborn.dev</li>
    <li>deepin.deepin-music</li>
    <li>deepin.deepin-ocr-plugin-manager</li>
    <li>deepin.deepin-pdfium</li>
    <li>deepin.deepin-picker</li>
    <li>deepin.deepin-pw-check</li>
    <li>deepin.deepin-screen-recorder</li>
    <li>deepin.deepin-screensaver</li>
    <li>deepin.deepin-service-manager</li>
    <li>deepin.deepin-shortcut-viewer</li>
    <li>deepin.deepin-system-monitor</li>
    <li>deepin.deepin-terminal</li>
    <li>deepin.deepin-turbo</li>
    <li>deepin.deepin-voice-note</li>
    <li>deepin.deepin-wallpapers</li>
    <li>deepin.dpa-ext-gnomekeyring</li>
    <li>deepin.dtkcommon</li>
    <li>deepin.dtkcore</li>
    <li>deepin.dtkcore.dev</li>
    <li>deepin.dtkcore.doc</li>
    <li>deepin.dtkdeclarative</li>
    <li>deepin.dtkdeclarative.dev</li>
    <li>deepin.dtkdeclarative.doc</li>
    <li>deepin.dtkgui</li>
    <li>deepin.dtkgui.dev</li>
    <li>deepin.dtkgui.doc</li>
    <li>deepin.dtkwidget</li>
    <li>deepin.dtkwidget.dev</li>
    <li>deepin.dtkwidget.doc</li>
    <li>deepin.dwayland</li>
    <li>deepin.image-editor</li>
    <li>deepin.qt5integration</li>
    <li>deepin.qt5platform-plugins</li>
    <li>deepin.startdde</li>
    <li>deepin.util-dfm</li>
  </ul>
</details>


From `deepin.deepin-clone`:
<details>
```
/build/source/app/src/main.cpp: In function 'int main(int, char**)':
/build/source/app/src/main.cpp:100:9: error: 'logger' was not declared in this scope; did you mean 'dlogger'?
  100 |         logger->registerAppender(rollingFileAppender);
      |         ^~~~~~
      |         dlogger
/build/source/app/src/main.cpp:179:5: error: 'logger' was not declared in this scope; did you mean 'dlogger'?
  179 |     logger->registerCategoryAppender("deepin.ghost", consoleAppender);
      |     ^~~~~~
      |     dlogger
```
</details>

From `deepin.deepin-reader`:

<details>
```
sidebar/BookMarkWidget.cpp: In member function 'void BookMarkWidget::initWidget()':
sidebar/BookMarkWidget.cpp:36:13: error: 'DGuiApplicationHelper' has not been declared
   36 |     connect(DGuiApplicationHelper::instance(), &DGuiApplicationHelper::themeTypeChanged, this, &BookMarkWidget::onUpdateTheme);
      |             ^~~~~~~~~~~~~~~~~~~~~
sidebar/BookMarkWidget.cpp:36:49: error: 'DGuiApplicationHelper' has not been declared
   36 |     connect(DGuiApplicationHelper::instance(), &DGuiApplicationHelper::themeTypeChanged, this, &BookMarkWidget::onUpdateTheme);
      |                                                 ^~~~~~~~~~~~~~~~~~~~~
sidebar/BookMarkWidget.cpp: In member function 'void BookMarkWidget::onUpdateTheme()':
sidebar/BookMarkWidget.cpp:193:5: error: 'DPalette' was not declared in this scope; did you mean 'Dtk::Gui::DPalette'?
  193 |     DPalette plt = Dtk::Gui::DGuiApplicationHelper::instance()->applicationPalette();
      |     ^~~~~~~~
      |     Dtk::Gui::DPalette
In file included from /nix/store/mq52m0jfc9abqhzlpx8i91ml30cg7yd7-dtkgui-5.6.17-dev/include/dtk5/DGui/DPalette:1,
                 from /nix/store/w439na2kjq604k7pkyf27h8299lb0w4y-dtkwidget-5.6.17-dev/include/dtk5/DWidget/dstyleditemdelegate.h:10,
                 from /nix/store/w439na2kjq604k7pkyf27h8299lb0w4y-dtkwidget-5.6.17-dev/include/dtk5/DWidget/DStyledItemDelegate:1,
                 from /nix/store/w439na2kjq604k7pkyf27h8299lb0w4y-dtkwidget-5.6.17-dev/include/dtk5/DWidget/dlistview.h:11,
                 from /nix/store/w439na2kjq604k7pkyf27h8299lb0w4y-dtkwidget-5.6.17-dev/include/dtk5/DWidget/DListView:1,
                 from sidebar/SideBarImageListview.h:9,
                 from sidebar/BookMarkWidget.cpp:8:
/nix/store/mq52m0jfc9abqhzlpx8i91ml30cg7yd7-dtkgui-5.6.17-dev/include/dtk5/DGui/dpalette.h:16:7: note: 'Dtk::Gui::DPalette' declared here
   16 | class DPalette : public QPalette
      |       ^~~~~~~~
sidebar/BookMarkWidget.cpp:194:5: error: 'plt' was not declared in this scope
  194 |     plt.setColor(Dtk::Gui::DPalette::Background, plt.color(Dtk::Gui::DPalette::Base));
      |     ^~~
```
</details>

wineee commented on pkgs/desktops/deepin/core/dde-application-manager/default.nix:
```diff
@@ -0,0 +1,64 @@
+    hash = "sha256-1P265xqlL/wML66nKdfTgkRx6MCpLwrt5rXu+CyeShU=";
+  };
+
+  # remove this patch after next release
```
dde-application-manager use `develop/refact` branch. The new application-manager does not have a hard coded path, but the interface has changed and needs to wait for other dde components to adapt

wineee committed:
deepin-reader: 6.0.2 -> 6.0.5

wineee committed:
deepin-clone: mark as broken

wineee committed:
deepin-screen-recorder: use a substituteInPlace candidate to patch

Co-authored-by: Peder Bergebakken Sundt <pbsds@hotmail.com>

wineee commented:
deepin-reader: fix build in 6.0.5

deepin-clone: mark as broken， Lack of maintenance in upstream, this package may be removed in the future

pbsds commented:
Error in `deepin.dde-gsettings-schemas`:

```
cp: missing destination file operand after '/nix/store/7zrawcajf0d58jamclfamzwn1f7mqpza-nixos-gsettings-desktop-schemas/share/gsettings-schemas/nixos-gsettings-overrides/glib-2.0/schemas'
Try 'cp --help' for more information.
```

wineee commented:
@ofborg build deepin.dde-gsettings-schemas

wineee committed:
dde-gsettings-schemas: Bring closer to GNOME

Same with https://github.com/NixOS/nixpkgs/commit/cb0c70b38855278096bfab28bcf2fcd69cc8b407

wineee commented:
@ofborg build deepin

wineee commented:
@ofborg build pantheon.elementary-gsettings-schemas

wineee commented:
@ofborg test deepin

pbsds commented:
Played around in a VM, everything i tried worked except the video player, but MPV had the same issue so it is likely the vm setup that did it.

<details>

![Screenshot from 2023-10-28 02-48-18](https://github.com/NixOS/nixpkgs/assets/140964/f916c1d9-3e2d-473c-8727-695697b372b3)
![Screenshot from 2023-10-28 02-48-40](https://github.com/NixOS/nixpkgs/assets/140964/727d4924-30d2-4b5c-8758-635e1f68e075)
![Screenshot from 2023-10-28 02-48-57](https://github.com/NixOS/nixpkgs/assets/140964/a0f08550-59a9-4452-b085-51010f2f5f87)
![Screenshot from 2023-10-28 02-49-13](https://github.com/NixOS/nixpkgs/assets/140964/686cb83b-b406-4fa7-803f-eda7c8066016)
![Screenshot from 2023-10-28 02-49-56](https://github.com/NixOS/nixpkgs/assets/140964/0c2c969e-26a8-4698-90b4-d2a40b18403f)
![Screenshot from 2023-10-28 02-50-23](https://github.com/NixOS/nixpkgs/assets/140964/4e9d9f81-719e-4e55-926d-95e3345e6ac2)
![Screenshot from 2023-10-28 02-50-52](https://github.com/NixOS/nixpkgs/assets/140964/62d711de-4fd9-4a58-a2f9-8a021611d2c2)
![Screenshot from 2023-10-28 02-51-30](https://github.com/NixOS/nixpkgs/assets/140964/9635c509-67ec-427c-97d1-a96df43ff177)
![Screenshot from 2023-10-28 02-55-02](https://github.com/NixOS/nixpkgs/assets/140964/dfd11b43-e5b8-4c87-b7c7-b69458782480)
![Screenshot from 2023-10-28 02-56-46](https://github.com/NixOS/nixpkgs/assets/140964/8c22adc4-57f4-4d6e-bb4d-62b7cb0a7bfb)
![Screenshot from 2023-10-28 02-56-53](https://github.com/NixOS/nixpkgs/assets/140964/0e70d4aa-42c5-48ed-8946-c2f261fc33fc)
![Screenshot from 2023-10-28 02-59-20](https://github.com/NixOS/nixpkgs/assets/140964/bec70ea5-126e-427d-88ea-de4e36884a0f)
![Screenshot from 2023-10-28 03-00-25](https://github.com/NixOS/nixpkgs/assets/140964/5eb8f92a-50f8-4948-8809-57783a350d06)
![Screenshot from 2023-10-28 03-00-40](https://github.com/NixOS/nixpkgs/assets/140964/db401a95-2363-4206-93de-366b4661786d)
![Screenshot from 2023-10-28 03-00-43](https://github.com/NixOS/nixpkgs/assets/140964/e0e2cf96-5a23-495c-a5ee-3bdddba99646)
![Screenshot from 2023-10-28 03-01-46](https://github.com/NixOS/nixpkgs/assets/140964/7f0d637f-51c0-494a-b966-8bc4747efa6c)
![Screenshot from 2023-10-28 03-02-06](https://github.com/NixOS/nixpkgs/assets/140964/f3a0ef6e-3850-4158-a276-b868ee7f2670)

</detail>

wegank commented:
Result of `nixpkgs-review pr 257400` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>deepin.deepin-clone</li>
  </ul>
</details>
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>deepin.dde-file-manager</li>
    <li>deepin.dde-session</li>
    <li>deepin.deepin-terminal</li>
    <li>deepin.dpa-ext-gnomekeyring</li>
    <li>deepin.startdde</li>
    <li>deepin.util-dfm</li>
  </ul>
</details>
<details>
  <summary>66 packages built:</summary>
  <ul>
    <li>deepin.dde-account-faces</li>
    <li>deepin.dde-api</li>
    <li>deepin.dde-app-services</li>
    <li>deepin.dde-appearance</li>
    <li>deepin.dde-application-manager</li>
    <li>deepin.dde-calendar</li>
    <li>deepin.dde-clipboard</li>
    <li>deepin.dde-control-center</li>
    <li>deepin.dde-control-center.dev</li>
    <li>deepin.dde-daemon</li>
    <li>deepin.dde-device-formatter</li>
    <li>deepin.dde-dock</li>
    <li>deepin.dde-dock.dev</li>
    <li>deepin.dde-launcher</li>
    <li>deepin.dde-launchpad</li>
    <li>deepin.dde-network-core</li>
    <li>deepin.dde-polkit-agent</li>
    <li>deepin.dde-qt-dbus-factory</li>
    <li>deepin.dde-session-shell</li>
    <li>deepin.dde-session-shell.dev</li>
    <li>deepin.dde-session-ui</li>
    <li>deepin.dde-widgets</li>
    <li>deepin.deepin-album</li>
    <li>deepin.deepin-calculator</li>
    <li>deepin.deepin-camera</li>
    <li>deepin.deepin-compressor</li>
    <li>deepin.deepin-desktop-schemas</li>
    <li>deepin.deepin-desktop-theme</li>
    <li>deepin.deepin-draw</li>
    <li>deepin.deepin-editor</li>
    <li>deepin.deepin-image-viewer</li>
    <li>deepin.deepin-kwin</li>
    <li>deepin.deepin-kwin.dev</li>
    <li>deepin.deepin-movie-reborn</li>
    <li>deepin.deepin-movie-reborn.dev</li>
    <li>deepin.deepin-music</li>
    <li>deepin.deepin-ocr-plugin-manager</li>
    <li>deepin.deepin-pdfium</li>
    <li>deepin.deepin-picker</li>
    <li>deepin.deepin-pw-check</li>
    <li>deepin.deepin-reader</li>
    <li>deepin.deepin-screen-recorder</li>
    <li>deepin.deepin-screensaver</li>
    <li>deepin.deepin-service-manager</li>
    <li>deepin.deepin-shortcut-viewer</li>
    <li>deepin.deepin-system-monitor</li>
    <li>deepin.deepin-turbo</li>
    <li>deepin.deepin-voice-note</li>
    <li>deepin.deepin-wallpapers</li>
    <li>deepin.dtkcommon</li>
    <li>deepin.dtkcore</li>
    <li>deepin.dtkcore.dev</li>
    <li>deepin.dtkcore.doc</li>
    <li>deepin.dtkdeclarative</li>
    <li>deepin.dtkdeclarative.dev</li>
    <li>deepin.dtkdeclarative.doc</li>
    <li>deepin.dtkgui</li>
    <li>deepin.dtkgui.dev</li>
    <li>deepin.dtkgui.doc</li>
    <li>deepin.dtkwidget</li>
    <li>deepin.dtkwidget.dev</li>
    <li>deepin.dtkwidget.doc</li>
    <li>deepin.dwayland</li>
    <li>deepin.image-editor</li>
    <li>deepin.qt5integration</li>
    <li>deepin.qt5platform-plugins</li>
  </ul>
</details>

wineee commented:
> Result of `nixpkgs-review pr 257400` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 
> 1 package marked as broken and skipped:
> 1 package blacklisted:
> 6 packages failed to build:
> 66 packages built:

I don't have an arm linux device, but [nixosTests.deepin on aarch64 Linux](https://logs.ofborg.org/?key=nixos/nixpkgs.257400&attempt_id=db823c89-fdcf-4a62-a7fd-f714c31c7362) success, could you please add more error log

wegank commented:
Result of `nixpkgs-review pr 257400` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>deepin.deepin-clone</li>
  </ul>
</details>
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>deepin.dde-file-manager</li>
    <li>deepin.dde-session</li>
    <li>deepin.deepin-terminal</li>
    <li>deepin.dpa-ext-gnomekeyring</li>
    <li>deepin.startdde</li>
    <li>deepin.util-dfm</li>
  </ul>
</details>
<details>
  <summary>66 packages built:</summary>
  <ul>
    <li>deepin.dde-account-faces</li>
    <li>deepin.dde-api</li>
    <li>deepin.dde-app-services</li>
    <li>deepin.dde-appearance</li>
    <li>deepin.dde-application-manager</li>
    <li>deepin.dde-calendar</li>
    <li>deepin.dde-clipboard</li>
    <li>deepin.dde-control-center</li>
    <li>deepin.dde-control-center.dev</li>
    <li>deepin.dde-daemon</li>
    <li>deepin.dde-device-formatter</li>
    <li>deepin.dde-dock</li>
    <li>deepin.dde-dock.dev</li>
    <li>deepin.dde-launcher</li>
    <li>deepin.dde-launchpad</li>
    <li>deepin.dde-network-core</li>
    <li>deepin.dde-polkit-agent</li>
    <li>deepin.dde-qt-dbus-factory</li>
    <li>deepin.dde-session-shell</li>
    <li>deepin.dde-session-shell.dev</li>
    <li>deepin.dde-session-ui</li>
    <li>deepin.dde-widgets</li>
    <li>deepin.deepin-album</li>
    <li>deepin.deepin-calculator</li>
    <li>deepin.deepin-camera</li>
    <li>deepin.deepin-compressor</li>
    <li>deepin.deepin-desktop-schemas</li>
    <li>deepin.deepin-desktop-theme</li>
    <li>deepin.deepin-draw</li>
    <li>deepin.deepin-editor</li>
    <li>deepin.deepin-image-viewer</li>
    <li>deepin.deepin-kwin</li>
    <li>deepin.deepin-kwin.dev</li>
    <li>deepin.deepin-movie-reborn</li>
    <li>deepin.deepin-movie-reborn.dev</li>
    <li>deepin.deepin-music</li>
    <li>deepin.deepin-ocr-plugin-manager</li>
    <li>deepin.deepin-pdfium</li>
    <li>deepin.deepin-picker</li>
    <li>deepin.deepin-pw-check</li>
    <li>deepin.deepin-reader</li>
    <li>deepin.deepin-screen-recorder</li>
    <li>deepin.deepin-screensaver</li>
    <li>deepin.deepin-service-manager</li>
    <li>deepin.deepin-shortcut-viewer</li>
    <li>deepin.deepin-system-monitor</li>
    <li>deepin.deepin-turbo</li>
    <li>deepin.deepin-voice-note</li>
    <li>deepin.deepin-wallpapers</li>
    <li>deepin.dtkcommon</li>
    <li>deepin.dtkcore</li>
    <li>deepin.dtkcore.dev</li>
    <li>deepin.dtkcore.doc</li>
    <li>deepin.dtkdeclarative</li>
    <li>deepin.dtkdeclarative.dev</li>
    <li>deepin.dtkdeclarative.doc</li>
    <li>deepin.dtkgui</li>
    <li>deepin.dtkgui.dev</li>
    <li>deepin.dtkgui.doc</li>
    <li>deepin.dtkwidget</li>
    <li>deepin.dtkwidget.dev</li>
    <li>deepin.dtkwidget.doc</li>
    <li>deepin.dwayland</li>
    <li>deepin.image-editor</li>
    <li>deepin.qt5integration</li>
    <li>deepin.qt5platform-plugins</li>
  </ul>
</details>

wegank commented:
`libsecret` fails on my machine, possibly unrelated.

wegank commented:
Result of `nixpkgs-review pr 257400` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>deepin.deepin-clone</li>
  </ul>
</details>
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>72 packages built:</summary>
  <ul>
    <li>deepin.dde-account-faces</li>
    <li>deepin.dde-api</li>
    <li>deepin.dde-app-services</li>
    <li>deepin.dde-appearance</li>
    <li>deepin.dde-application-manager</li>
    <li>deepin.dde-calendar</li>
    <li>deepin.dde-clipboard</li>
    <li>deepin.dde-control-center</li>
    <li>deepin.dde-control-center.dev</li>
    <li>deepin.dde-daemon</li>
    <li>deepin.dde-device-formatter</li>
    <li>deepin.dde-dock</li>
    <li>deepin.dde-dock.dev</li>
    <li>deepin.dde-file-manager</li>
    <li>deepin.dde-launcher</li>
    <li>deepin.dde-launchpad</li>
    <li>deepin.dde-network-core</li>
    <li>deepin.dde-polkit-agent</li>
    <li>deepin.dde-qt-dbus-factory</li>
    <li>deepin.dde-session</li>
    <li>deepin.dde-session-shell</li>
    <li>deepin.dde-session-shell.dev</li>
    <li>deepin.dde-session-ui</li>
    <li>deepin.dde-widgets</li>
    <li>deepin.deepin-album</li>
    <li>deepin.deepin-calculator</li>
    <li>deepin.deepin-camera</li>
    <li>deepin.deepin-compressor</li>
    <li>deepin.deepin-desktop-schemas</li>
    <li>deepin.deepin-desktop-theme</li>
    <li>deepin.deepin-draw</li>
    <li>deepin.deepin-editor</li>
    <li>deepin.deepin-image-viewer</li>
    <li>deepin.deepin-kwin</li>
    <li>deepin.deepin-kwin.dev</li>
    <li>deepin.deepin-movie-reborn</li>
    <li>deepin.deepin-movie-reborn.dev</li>
    <li>deepin.deepin-music</li>
    <li>deepin.deepin-ocr-plugin-manager</li>
    <li>deepin.deepin-pdfium</li>
    <li>deepin.deepin-picker</li>
    <li>deepin.deepin-pw-check</li>
    <li>deepin.deepin-reader</li>
    <li>deepin.deepin-screen-recorder</li>
    <li>deepin.deepin-screensaver</li>
    <li>deepin.deepin-service-manager</li>
    <li>deepin.deepin-shortcut-viewer</li>
    <li>deepin.deepin-system-monitor</li>
    <li>deepin.deepin-terminal</li>
    <li>deepin.deepin-turbo</li>
    <li>deepin.deepin-voice-note</li>
    <li>deepin.deepin-wallpapers</li>
    <li>deepin.dpa-ext-gnomekeyring</li>
    <li>deepin.dtkcommon</li>
    <li>deepin.dtkcore</li>
    <li>deepin.dtkcore.dev</li>
    <li>deepin.dtkcore.doc</li>
    <li>deepin.dtkdeclarative</li>
    <li>deepin.dtkdeclarative.dev</li>
    <li>deepin.dtkdeclarative.doc</li>
    <li>deepin.dtkgui</li>
    <li>deepin.dtkgui.dev</li>
    <li>deepin.dtkgui.doc</li>
    <li>deepin.dtkwidget</li>
    <li>deepin.dtkwidget.dev</li>
    <li>deepin.dtkwidget.doc</li>
    <li>deepin.dwayland</li>
    <li>deepin.image-editor</li>
    <li>deepin.qt5integration</li>
    <li>deepin.qt5platform-plugins</li>
    <li>deepin.startdde</li>
    <li>deepin.util-dfm</li>
  </ul>
</details>

pbsds commented:
Do you intend to push any more changes to this?

wineee commented:
> Do you intend to push any more changes to this?


I have rebased for https://github.com/NixOS/nixpkgs/pull/264280 , no other changes

wineee commented:
rebased to reslove merge conflict

K900 commented:
We don't want to ship a beta version of DDE in 23.11, right?

wineee commented:
> We don't want to ship a beta version of DDE in 23.11, right?

right, It's better to ship it in 24.05-unstable, allow enough time for testing and improvement
(DDE has updated many components after the release of v23Beta. the beta version more refers to deepin os, and DDE has not yet an independent version.)

wineee committed:
dde-launcher: remove

wineee committed:
dde-launchpad: 0.0.2 -> 0.2.1

wineee committed:
deepin(go-package): use `vendorHash` instead of `vendorSha256`

wineee commented:
After NixOS-23.11 branch-off,   this pr is ready for review now

![image](https://github.com/NixOS/nixpkgs/assets/22803888/c60a49f4-d09f-4699-8783-3c2265792c01)
